### PR TITLE
Schema/expression language

### PR DESCRIPTION
**Pull request by rwblair**
- Introduces functions for schema expression language outlined [here](https://github.com/bids-standard/bids-specification/tree/master/src/schema#Expressions).
- ~~Adds nifti header loading but should revised. Causes a considerable slow down from loading whole nifti files synchronously.~~
- Adds fully loading events.tsv for associations checks.

### CODE REVIEW

**Comment by rwblair** [comment]
Need to add tests. Expression testing examples could come from specification itself a la bids-standard/bids-specification#1274

File: bids-validator/src/files/nifti.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import 'https://raw.githubusercontent.com/rii-mango/NIFTI-Reader-JS/master/release/current/nifti-reader-min.js'
+import { BIDSFileDeno } from './deno.ts'
+
+export function loadHeader(file: BIDSFile) {
```
Comment (nellh) [inline bids-validator/src/files/nifti.ts]:
```suggestion
export function loadHeader(file: BIDSFileDeno) {
```

File: bids-validator/src/files/nifti.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  const header = nifti.readHeader(Deno.readFileSync(file._getPath()).buffer)
```
Comment (nellh) [inline bids-validator/src/files/nifti.ts]:
```suggestion
  const buf = file.readBytes(1024)
  const header = globalThis.nifti.readHeader(buf.buffer)
```

Added this in #1536 to keep this from requiring the private path method.

File: bids-validator/src/schema/context.ts [hunk 3]
```diff
@@ -124,8 +147,11 @@ export class BIDSContext implements Context {
   }
 
   async asyncLoads() {
-    await this.loadSidecar()
-    await this.loadColumns()
-    await this.loadAssociations()
+    await Promise.all([
```
Comment (nellh) [inline bids-validator/src/schema/context.ts]:
```suggestion
    await Promise.allSettled([
```

**Review by nellh — CHANGES_REQUESTED** [review]

**Review by nellh — APPROVED** [review]

**Pull request merged**